diff --git a/grpc/tools/buildgen/_mako_renderer.py b/grpc/tools/buildgen/_mako_renderer.py
index 77c3c49f..aa4de699 100755
--- a/grpc/tools/buildgen/_mako_renderer.py
+++ b/grpc/tools/buildgen/_mako_renderer.py
@@ -113,7 +113,7 @@ def main(argv: List[str]) -> None:
     for arg in args:
         got_input = True
         with open(arg) as f:
-            srcs = list(yaml.load_all(f.read(), Loader=yaml.FullLoader))
+            srcs = list(yaml.safe_load_all(f.read(), Loader=yaml.FullLoader))
         for src in srcs:
             if isinstance(src, str):
                 assert len(srcs) == 1
diff --git a/grpc/tools/buildgen/generate_projects.py b/grpc/tools/buildgen/generate_projects.py
index 2e64484f..9515df12 100755
--- a/grpc/tools/buildgen/generate_projects.py
+++ b/grpc/tools/buildgen/generate_projects.py
@@ -72,7 +72,7 @@ def preprocess_build_files() -> _utils.Bunch:
     for build_file in args.build_files:
         with open(build_file, 'r') as f:
             _utils.merge_json(build_spec,
-                              yaml.load(f.read(), Loader=yaml.FullLoader))
+                              yaml.safe_load(f.read(), Loader=yaml.FullLoader))
     # Executes plugins. Plugins update the build spec in-place.
     for py_file in sorted(glob.glob('tools/buildgen/plugins/*.py')):
         plugin = _utils.import_python_module(py_file)
